tyr/servers/server.py

```diff
@@ -597,74 +617,89 @@ def resolve_iam_role(self):
+                        continue  # Go to the next policy
 
-                tyr_copy = json.loads(policies[policy])
+                    self.log.info('Policy "{policy}" does not exist'.format(
+                                  policy=policy))
 
-                aws_copy = self.iam.get_role_policy(self.role, policy)
-                aws_copy = aws_copy['get_role_policy_response']
-                aws_copy = aws_copy['get_role_policy_result']
-                aws_copy = aws_copy['policy_document']
-                aws_copy = urllib.unquote(aws_copy)
-                aws_copy = json.loads(aws_copy)
+                    try:
+                        self.iam.put_role_policy(self.role, policy,
+                                                 rolePolicy)
 
-                if tyr_copy == aws_copy:
-                    self.log.info('Policy "{policy}" is accurate'.format(
-                                  policy=policy))
+                        self.log.info('Added policy "{policy}"'.format(
+                                      policy=policy))
+                    except Exception as e:
+                        self.log.error(str(e))
+                        raise e
 
                 else:
 
-                    self.log.warn('Policy "{policy}" has been modified'.format(
+                    self.log.info('Policy "{policy}" already exists'.format(
                                   policy=policy))
 
-                    try:
-                        self.iam.delete_role_policy(self.role, policy)
+                    tyr_copy = json.loads(policies[policy])
+
+                    aws_copy = self.iam.get_role_policy(self.role, policy)
+                    aws_copy = aws_copy['get_role_policy_response']
+                    aws_copy = aws_copy['get_role_policy_result']
+                    aws_copy = aws_copy['policy_document']
+                    aws_copy = urllib.unquote(aws_copy)
+                    aws_copy = json.loads(aws_copy)
 
-                        self.log.info('Removed policy "{policy}"'.format(
+                    if tyr_copy == aws_copy:
+                        self.log.info('Policy "{policy}" is accurate'.format(
                                       policy=policy))
-                    except Exception as e:
-                        self.log.error(str(e))
-                        raise e
 
-                    try:
-                        self.iam.put_role_policy(self.role, policy,
-                                                 policies[policy])
+                    else:
 
-                        self.log.info('Added policy "{policy}"'.format(
+                        self.log.warn('Policy "{policy}" has been modified'.format(
                                       policy=policy))
-                    except Exception as e:
-                        self.log.error(str(e))
-                        raise e
+
+                        try:
+                            self.iam.delete_role_policy(self.role, policy)
```

Comment:
so you're deleting the policy then adding it again?  That would cause a brief moment when that policy is removed from a server if there's any using that IAM role.  This could potentially cause production issues.